src/util/xml_to_json.cc

```diff
@@ -113,11 +114,23 @@ std::string Xml2Json::getAsString(const char* str)
     return "\"" + escape(str, '\\', '"') + '"';
 }
 
-std::string Xml2Json::getValue(const char* text)
+std::string Xml2Json::getValue(const char* name, const char* text, const Hints* hints)
 {
-    // type information is gone, we need to guess...
+    std::string item = name;
     std::string str = text;
 
+    for (const auto& hint : hints->asType) {
```

Comment:
If hints can't be null and it's not resigned, maybe make it a reference?

Also nice opportunity to try c++17's structured bindings `const auto& [name, type] : hints->asType`
(You don't need to create a string `item`. sts::string can compare with char*).

Reply:
Agreed on the reference. Indifferent on the bind

Reply:
I was sleepy yesterday... If this is a `std::map`, there is no need to iterate over it linearly. Use `find`.

Reply:
find_if is not what i meant. std::map has `find` method which is logarithmic. The search is by the key, right?

Reply:
Well, as I said, I copied. I am releaning C++ after 20 years of coding C# 👍

Reply:
Learn 20 then. Ranges (not linq, but close), coroutines (async, await, although only language feature, no support on standard library), concepts (close to C# generics in some way...). 😂